### PR TITLE
[Feature] Adding Feature.md/BugFix.md Pull Request Template.

### PR DESCRIPTION
**Pull request by Leopere**
# Scope
Adding a BugFix.md pull request template along with a Feature.md one.  Also added basic CONTRIBUTING.md guidelines for now.  We can revise at a later date if nobody has any disputes with what we have in this now.

Additional tweak to FUNDING.yml as well.  Removed LibrePay as we're not using that anymore.

## Closes/Fixes Issues
none

## Platforms tested
* [ ] Windows 10
* [x] OS X Catalina
* [ ] Linux
* [ ] Arm

## Other testing QA Notes
This PR has been in discussion within the #code-review channel on Discord.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+name: Bug Fix
+about: Add a bug fix this project!
+title: "[Feature] Short Form Title (50 chars or less.)"
```
Comment (WithoutPants) [inline .github/PULL_REQUEST_TEMPLATE/BugFix.md]: This should be `[Bug Fix]`.

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+<!-- Declare any issues by typing `fixes #1` or `closes #1` for example so that the automation can kick in when this is merged -->
+## This Pull Request Attempts to close the following issues
+
+<!-- Please put an X in the following items to show that you've done your due diligence. -->
+## Due Diligence Checklist
+
+* [ ] I've run Code linters and [gofmt](https://golang.org/cmd/gofmt/) to make sure that my code is readable.
+* [ ] I have read through formerly submitted pull requests and issues to make sure that this is required and so that I can manage to close any Issues needing closed relating to this feature submission.
+* [ ] To the best of my ability, I've followed the [CONTRIBUTING.md](/CONTRIBUTING.md) to make sure that I'm up to date on how to best contribute to Stash projects.
+* [ ] I  commented adequately on my code with the expectation in mind that anyone else should be able to look at this code I've submitted and know exactly what's happening and what the expectations are.
```
Comment (WithoutPants) [inline .github/PULL_REQUEST_TEMPLATE/BugFix.md]: In my opinion, this is way too wordy. I think these should be as terse as possible, with the details of this stuff documented in `CONTRIBUTING.md` file (which I also note does not currently exist).

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+* [ ] I acknowledge that if applicable to me, submitting and subsequent acceptance of this Pull Request I, the code contributor of this Pull Request, agree and acknowledge my understanding that the new code license has now been updated to [AGPL](/LICENSE.md). I agree that all code before this Pull Request, which I've previously submitted, is now to be re-licensed under the new license AGPL and no longer the former MIT license.
```
Comment (WithoutPants) [inline .github/PULL_REQUEST_TEMPLATE/BugFix.md]: I think this is redundant and unnecessary. The license is changed for subsequent pull requests and contributors are expected to know this when making pull requests.

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  <!-- Explain why you may not have checked all of the above boxes. -->
+  ### In case I haven't checked all of the boxes above here is my explanation as to why not.
```
Comment (WithoutPants) [inline .github/PULL_REQUEST_TEMPLATE/BugFix.md]: I'd remove the heading here and just leave the comment, adding an "if applicable".

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+
+<!-- Please check off what platforms that you've tested your build on so that we can potentially skip some steps in accepting your Pull Request -->
+## Platforms that I've tested this build with the following platforms.
+* [ ] Windows 10
+* [ ] OS X Catalina
+* [ ] Linux
+* [ ] Arm
+
```
Comment (WithoutPants) [inline .github/PULL_REQUEST_TEMPLATE/BugFix.md]: I'd also like to see something about what has been tested by the submitter, and what possible impacts/areas there are that may need retesting by others.

A checkbox is probably unnecessary for platforms, just leave it for the user to fill in. I'd probably make the heading more terse as well - "Platforms tested against" or something.

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md (hunk 2 above)
Comment (Leopere) [reply .github/PULL_REQUEST_TEMPLATE/BugFix.md]: We could add some of the lengthiness moved over to the CONTRIBUTING.md instead.  I was avoiding adding a CONTRIBUTING.md file so I suppose I'll have to add that here too.

**Comment by bnkai** [comment]
@Leopere 
i don't know if i misread this 
`I have read through formerly submitted pull requests and issues to make sure that this is required and so that I can manage to close any issues needing closed relating to this feature submission.`
When reading this i get the impression that the submitter has to fix/implement something already mentioned/required. Is it implied that if you want to implement a new feature you should first make an issue before a PR?

**Comment by Leopere** [comment]
> @Leopere
> i don't know if i misread this
> `I have read through formerly submitted pull requests and issues to make sure that this is required and so that I can manage to close any issues needing closed relating to this feature submission.`
> When reading this i get the impression that the submitter has to fix/implement something already mentioned/required. Is it implied that if you want to implement a new feature you should first make an issue and then a PR?

I'm open to rephrasing however I just want to suggest that the code contributor looks through to see if they're duplicating any work or can close any issues with their pull requests.

**Comment by Leopere** [comment]
@bnkai I've reworded and repositioned that checklist over to the newly created [CONTRIBUTING.md](./docs/CONTRIBUTING.md) file.

File: docs/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+
+## Technical Debt
+Please be sure to consider how heavily your contribution impacts the maintainability of the project long term, sometimes less is more.  We don't want to merge collossal pull requests with hundreds of dependencies by a driveby contributor.
+
+## Due Diligence Checklist
+Please make sure that you've considered the following before you submit your Pull Requests as ready for merging.
+* I've run Code linters and [gofmt](https://golang.org/cmd/gofmt/) to make sure that my code is readable.
+* I have read through formerly submitted pull requests and git issues issues to make sure that this contribution is required and isn't a duplicate. Also, so that I can manage to close any git Issues needing closed relating to this feature submission.
```
Comment (bnkai) [inline docs/CONTRIBUTING.md]: Typo `git issues issues` and `git Issues`
probably github issues

File: docs/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+* To the best of my ability, I've followed the [CONTRIBUTING.md](./CONTRIBUTING.md) to make sure that I'm up to date on how to best contribute to Stash projects.
```
Comment (bnkai) [inline docs/CONTRIBUTING.md]: The contributor is already reading the CONTRIBUTING.md is this needed?

Comment (Leopere) [reply docs/CONTRIBUTING.md]: Likely not I just copied over the points and edited them a little.

File: docs/CONTRIBUTING.md (hunk 6 above)
Comment (Leopere) [reply docs/CONTRIBUTING.md]: good catch

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md (hunk 5 above)
Comment (WithoutPants) [reply .github/PULL_REQUEST_TEMPLATE/BugFix.md]: > A checkbox is probably unnecessary for platforms, just leave it for the user to fill in. I'd probably make the heading more terse as well - "Platforms tested against" or something.

This doesn't look like it's been resolved. Having checkboxes in the template also adds a tasks icon to the pull request in the list page, which I think is incorrect.

File: docs/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+* I have read through formerly submitted [pull requests](https://github.com/stashapp/stash/pulls) and [git issues](https://github.com/stashapp/stash/issues) to make sure that this contribution is required and isn't a duplicate. Also, so that I can manage to close any git Issues needing closed relating to this feature submission.
+* I  commented adequately on my code with the expectation in mind that anyone else should be able to look at this code I've submitted and know exactly what's happening and what the expectations are.
+
+### Legal Agreements
+* I acknowledge that if applicable to me, submitting and subsequent acceptance of this Pull Request I, the code contributor of this Pull Request, agree and acknowledge my understanding that the new code license has now been updated to [AGPL](/LICENSE.md). I agree that all code before this Pull Request, which I've previously submitted, is now to be re-licensed under the new license AGPL and no longer the former MIT license.
+
+  <!-- Explain why you may not have checked all of the above boxes. -->
+  **In case I haven't checked all of the boxes above here is my explanation as to why not in your Pull Request.**
+
```
Comment (WithoutPants) [inline docs/CONTRIBUTING.md]: Looks like a copy/paste error from the other template.

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+name: Bug Fix
+about: Add a bug fix this project!
+title: "[Bug Fix] Short Form Title (50 chars or less.)"
+labels: bug
+assignees: 'WithoutPants, bnkai, Leopere'
+
+---
+<!-- Please make sure to read https://github.com/stashapp/stash/docs/CONTRIBUTING.md and check that you understand and have followed it as best as possible -->
+# Pre-Submission Checklist
+* [ ] I have read [CONTRIBUTING.md](/docs/CONTRIBUTING.md)
```
Comment (WithoutPants) [inline .github/PULL_REQUEST_TEMPLATE/BugFix.md]: Requiring the user to tick this box is unnecessary, and as mentioned elsewhere, adds tasks to the pull request entry in the pull request page. The comment is sufficient.

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md (hunk 5 above)
Comment (Leopere) [reply .github/PULL_REQUEST_TEMPLATE/BugFix.md]: I thought about it for a bit and figured that the checkboxes might imply the platforms that we support vs other platforms that are supported "by the community" perhaps?  Such as these NAS setups and whatnot.

Comment (WithoutPants) [reply .github/PULL_REQUEST_TEMPLATE/BugFix.md]: It's sufficient to leave that in a comment, then the contributor can copy and paste as needed.

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md (hunk 9 above)
Comment (Leopere) [reply .github/PULL_REQUEST_TEMPLATE/BugFix.md]: Fixing in a new commit

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md (hunk 5 above)
Comment (Leopere) [reply .github/PULL_REQUEST_TEMPLATE/BugFix.md]: I'm not sure but this change requested just isn't sitting right with me. 

There are other fringe benefits as well to this where reviewers can potentially tick off their verified platforms.

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md (hunk 9 above)
Comment (Leopere) [reply .github/PULL_REQUEST_TEMPLATE/BugFix.md]: Should be removed properly.

File: .github/PULL_REQUEST_TEMPLATE/BugFix.md (hunk 5 above)
Comment (WithoutPants) [reply .github/PULL_REQUEST_TEMPLATE/BugFix.md]: image

This is the issue that I'm referring to. It is a misuse of the tasks functionality. Review platforms are better communicated via explicit comment, rather than anonymous ticking of checkboxes. 

I'd want to hear more benefits of this method of showing tested platforms - to my mind, the benefits given are insufficient.

Comment (Leopere) [reply .github/PULL_REQUEST_TEMPLATE/BugFix.md]: conceded
